src/book/service.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  BadRequestException,
+  Injectable,
+  InternalServerErrorException,
+} from '@nestjs/common';
+import { PrismaService } from 'src/prisma/service';
+import { BookDTO } from './dtos';
+
+@Injectable()
+export class BookService {
+  constructor(private readonly prisma: PrismaService) {}
+  async createBook(
+    dto: BookDTO,
+    authorId: number,
+    categories: { id: number }[],
+  ) {
+    try {
+      const data = { ...dto };
+      if (authorId) {
+        data['author'] = {
+          connect: {
+            id: authorId,
+          },
+        };
+      }
+      if (Array.isArray(categories) && categories?.length > 0) {
+        data['categories'] = {
+          connect: categories,
+        };
+      }
+      return await this.prisma.book.create({
+        data,
+        include: {
+          author: true,
+          categories: true,
+        },
+      });
+    } catch (error) {
+      console.log(error.message);
```

Comment:
Dont use console log in nest.
Add logger at the beginning of the file, then use logger to log error.
Error message should be logger.error instead of normal log
```suggestion
      this.logger.error(error.message);
```